File: backend/app/schemas/application.py
from pydantic import BaseModel
from typing import List, Optional

class SectionSchema(BaseModel):
    id: str
    title: str
    prompt: str
    content: str
    aiSuggestedContent: Optional[str] = None
    keyPoints: Optional[List[str]] = None
    wordCountLimit: Optional[int] = None

class DocumentStatusSchema(BaseModel):
    name: str
    isUploaded: bool

class ApplicationSchema(BaseModel):
    id: str
    scholarshipId: str
    scholarshipName: str
    provider: str
    deadline: str
    amount: float
    status: str
    completionPercentage: int
    sections: List[SectionSchema]
    requiredDocuments: List[DocumentStatusSchema]
    updatedAt: str
    version: int = 1

File: frontend/src/types/application.ts
export type ApplicationStatus = 'draft' | 'in_progress' | 'ready_for_review' | 'submitted';

export interface ApplicationSection {
  id: string;
  title: string;
  prompt: string;
  content: string;
  aiSuggestedContent?: string;
  keyPoints?: string[];
  wordCountLimit?: number;
}

export interface Application {
  id: string;
  scholarshipId: string;
  scholarshipName: string;
  provider: string;
  deadline: string;
  amount: number;
  status: ApplicationStatus;
  completionPercentage: number;
  sections: ApplicationSection[];
  requiredDocuments: { name: string; isUploaded: boolean }[];
  updatedAt: string;
  version: number;
}
